internal/cli/cmd/connectivity.go

```diff
@@ -65,6 +69,34 @@ func newCmdConnectivityTest() *cobra.Command {
 				}
 			}
 
+			// Validate toleration string
+			var allowedEffects = []string{"NoSchedule", "NoExecute", "PreferNoSchedule"}
+			for _, toleration := range podTolerations {
+				tolerationString := strings.Split(toleration, ":")
+				if len(tolerationString) < 2 {
+					return fmt.Errorf("invalid format: %s, toleration string should be of format key1=value1:effect", toleration)
+				}
+				validEffect := false
+				for _, e := range allowedEffects {
+					if tolerationString[1] == e {
+						validEffect = true
```

Comment:
As @EricMountain [suggested in #1](https://github.com/DataDog/cilium-cli/pull/1/files#r646717628) & given that we now seem to be compiling with go >= 1.18 we can probably leverage the new "slice contains" generic for this assertion? 🤔 https://pkg.go.dev/golang.org/x/exp/slices#Contains